datahub-web-react/src/app/permissions/policy/policyUtils.ts

```diff
@@ -126,3 +127,150 @@ export const setFieldValues = (
+            resolvedGroups: newPolicy?.actors?.resolvedGroups || [],
+            resolvedRoles: newPolicy?.actors?.resolvedRoles || [],
+        },
+        resources : {
+            ...newPolicy.resources,
+            type: newPolicy?.resources?.type || '', // replace '' with the default value for type
+            allResources: newPolicy?.resources?.allResources || false,
+            resources: newPolicy?.resources?.resources || [],
+        }
+    };
+
+    // Add our new policies into the existing list.
+    const newPolicies = [completeNewPolicy, ...(currData?.listPolicies?.policies || [])];
+    
+    // Write our data back to the cache.
+    client.writeQuery({
+        query: ListPoliciesDocument,
+        variables: {
+            input: {
+                start: 0,
+                count: pageSize,
+                query,
+            },
+        },
+        data: {
+            listPolicies: {
+                start: 0,
+                count: (currData?.listPolicies?.count || 0) + 1,
+                total: (currData?.listPolicies?.total || 0) + 1,
+                policies: newPolicies,
+            },
+        },
+    });
+};
+
+
+/**
+ * Remove an entry from the ListPolicies cache.
+ */
+export const removeFromListPoliciesCache = (client, urn, page, pageSize, query) => {
+    // Read the data from our cache for this query.
+    const currData: ListPoliciesQuery | null = client.readQuery({
+        query: ListPoliciesDocument,
+        variables: {
+            input: {
+                start: (page - 1) * pageSize,
+                count: pageSize,
+                query,
+            },
+        },
+    });
+
+    const newPolicies = [
+        ...(currData?.listPolicies?.policies || []).filter((policy) => policy.urn !== urn),
+    ];
+
+    // Write our data back to the cache.
+    client.writeQuery({
+        query: ListPoliciesDocument,
+        variables: {
+            input: {
+                start: (page - 1) * pageSize,
+                count: pageSize,
+                query,
+            },
+        },
+        data: {
+            listPolicies: {
+                start: currData?.listPolicies?.start || 0,
+                count: (currData?.listPolicies?.count || 1) - 1,
+                total: (currData?.listPolicies?.total || 1) - 1,
+                policies: newPolicies,
+            },
+        },
+    });
+};
+
+
+/**
+ * update an entry from the ListPolicies cache.
+ */
+export const updateListPoliciesCache = (client, updatedPolicy, pageSize, query) => {
+    // Read the data from our cache for this query.
+    const currData: ListPoliciesQuery = client.readQuery({
+        query: ListPoliciesDocument,
+        variables: {
+            input: {
+                start: 0,
+                count: pageSize,
+                query,
+            },
+        },
+    });
+
+    const updatedPolicies = [
+        ...(currData?.listPolicies?.policies || []).filter((policy) => 
```

Comment:
I believe this should be a map and not a "filter".

Also, don't we need to do the same thing as above where we take the policy and map it into the format expected by apollo cache? 

i think this could be causing the failures we are seeing